build.gradle.kts

```diff
@@ -0,0 +1,339 @@
+    }
+}
+
+val copyArtifacts = tasks.register("copyArtifacts", Copy::class) {
+    dependsOn(tasks.build)
+
+    val jarArchivePath = project.tasks.jar.get().archiveFile.get().asFile.path
+    val exeFilePath = jarArchivePath.replace(".jar", ".exe").replace("libs", "launch4j")
+    val zipFilePath = jarArchivePath.replace(".jar", ".zip").replace("libs", "distributions")
+
+    from(project.tasks.shadowJar.get().archiveFile)
+    from(exeFilePath)
+    from(zipFilePath)
+
+    into("$projectDir/dist")
+}
+
+val downloadNewerUniversalJavaApplicationStub =
+    tasks.register<de.undercouch.gradle.tasks.download.Download>("downloadNewerUniversalJavaApplicationStub") {
+        description = "Downloads newer universalJavaApplicationStub"
+        src("https://raw.githubusercontent.com/tofi86/universalJavaApplicationStub/404f5c1b008d6296065de7a93406b387c9f3dce1/src/universalJavaApplicationStub")
+        dest(file("$buildDir/macApp/${project.name}.app/Contents/MacOS/universalJavaApplicationStub"))
+        overwrite(true)
+    }
+
+
+tasks.register("createTestLauncherDir") {
+    project.file("testLauncher/dev").mkdirs()
+}
+
+val createMacApp = tasks.register<Zip>("createMacApp") {
+    dependsOn(tasks.createApp, tasks.shadowJar, downloadNewerUniversalJavaApplicationStub)
+    from("$buildDir/macApp") {
+        include("${project.name}.app/**")
+        exclude("${project.name}.app/Contents/MacOS")
+    }
+    archiveFileName = "${project.name}-${project.version}.zip"
+}
+
+val printCopyArtifactsFinished = tasks.register("printCopyArtifactsFinished") {
+    println("ATLauncher has been built. Distribution files are located in the dist directory.")
+}
+copyArtifacts.get().finalizedBy(printCopyArtifactsFinished)
+
+tasks.clean {
+    doFirst {
+        println("Deleting `$projectDir/dist`")
+        delete("$projectDir/dist")
+    }
+}
+
+project.afterEvaluate {
+    tasks.check {
+        /*
+        Trick gradle by not "removing" a task
+         (which throws an unsupported error),
+         Instead filter current tasks and then set the new list.
+        */
+        setDependsOn(
+            dependsOn.filterNot {
+                (it as? Task)?.name == tasks.checkLicenses.name
+            }
+        )
+    }
+}
+
+val shouldIgnoreUpdate: (String) -> Boolean = { version ->
+    listOf("ALPHA", "BETA", "RC", "-M").any { version.uppercase().contains(it) }
+}
+
+tasks.dependencyUpdates.configure {
+    rejectVersionIf {
+        shouldIgnoreUpdate(candidate.version)
+    }
+}
+
+tasks.build.get().finalizedBy(copyArtifacts)
+tasks.build.get().finalizedBy(copyArtifacts)
+tasks.shadowJar.get().dependsOn(tasks.jar)
+tasks.build.get().dependsOn(tasks.createExe, createMacApp)
+tasks.startScripts.get().dependsOn(tasks.shadowJar)
+tasks.createExe.get().dependsOn(tasks.shadowJar)
+tasks.createAppZip.get().dependsOn(downloadNewerUniversalJavaApplicationStub)
+tasks.createDmg.get().dependsOn(downloadNewerUniversalJavaApplicationStub)
```

Comment:
why not use `tasks{}` at this point?

Reply:
It wasn't in the old Groovy code and I didn't want to make too many changes that will result in getting the PR rejected.

Reply:
If it would make it clearer/more compliant to best practices, I'm all for doing it